OpenRA.Mods.Common/Traits/SupportPowers/SelectDirectionalTarget.cs

```diff
@@ -106,28 +110,38 @@ void IOrderGenerator.Tick(World world)
 
 		bool IsOutsideDragZone
 		{
-			get { return dragStarted && (dragLocation - location).Length > 20; }
+			get { return dragStarted && (dragLocation - targetLocation).Length > 20; }
 		}
 
 		IEnumerable<IRenderable> IOrderGenerator.Render(WorldRenderer wr, World world) { yield break; }
 
 		IEnumerable<IRenderable> IOrderGenerator.RenderAboveShroud(WorldRenderer wr, World world)
 		{
-			if (!beginDrag)
+			if (!activated)
 				return Enumerable.Empty<IRenderable>();
 
-			var palette = wr.Palette("chrome");
+			var targetPalette = wr.Palette(targetPlaceholderCursorPalette);
+
+			var location = activated ? targetLocation : Viewport.LastMousePos;
 			var worldPx = wr.Viewport.ViewToWorldPx(location);
 			var worldPos = wr.ProjectedPosition(worldPx);
-			var renderables = new List<IRenderable>(targetCursor.Render(worldPos, WVec.Zero, -511, palette, 1 / wr.Viewport.Zoom));
+			var renderables = new List<IRenderable>(targetCursor.Render(worldPos, WVec.Zero, -511, targetPalette, 1 / wr.Viewport.Zoom));
 
 			if (IsOutsideDragZone)
-				renderables.Add(new SpriteRenderable(currentArrow.Sprite, worldPos, WVec.Zero, -511, palette, 1 / wr.Viewport.Zoom, true));
+			{
+				var directionPalette = wr.Palette(directionArrowPalette);
+				renderables.Add(new SpriteRenderable(currentArrow.Sprite, worldPos, WVec.Zero, -511, directionPalette, 1 / wr.Viewport.Zoom, true));
+			}
 
 			return renderables;
 		}
 
-		string IOrderGenerator.GetCursor(World world, CPos cell, int2 worldPixel, MouseInput mi) { return beginDrag ? "invisible" : cursor; }
+		string IOrderGenerator.GetCursor(World world, CPos cell, int2 worldPixel, MouseInput mi) { return cursor; }
+
+		void IOrderGenerator.Deactivate()
+		{
+			Game.HideCursor = false;
```

Comment:
Probably a good idea to add an `if (activated)` here in case we implement more things that hide the cursor.